### PR TITLE
NMS-12814: Interfaces incorrectly marked as having flows resulting in no data via Helm

### PR DESCRIPTION
**Pull request by patrick-schweizer**
Jira https://issues.opennms.org/browse/NMS-12814

### CODE REVIEW

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java [hunk 1]
```diff
@@ -221,11 +236,81 @@ public void testNMS12740() throws Exception {
...
+            checkInterfaces(1, "192.168.1.1");
+            checkInterfaces(2);
+
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.INGRESS)), getMockFlowSource());
+            checkInterfaces(1, "192.168.1.1", "192.168.1.1");
+            checkInterfaces(2);
+
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.INGRESS)), getMockFlowSource());
+            checkInterfaces(1, "192.168.1.1", "192.168.1.1");
+            checkInterfaces(2);
```
Comment (patrick-schweizer) [inline features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:293]: @fooker , @christianpape, the test will succeed with this implementation. But It is not clear to me why the system behaves like it does. Why is it green when I implement it that way?

I assume the checkInterfaces method is implemented wrong. I don't have enough domain knowledge to understand what we need to test for...
Maybe you can help me out here... Thanks!

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java [hunk 2]
```diff
@@ -221,11 +232,97 @@ public void testNMS12740() throws Exception {
...
+
+            Integer ingress = 2;
+            Integer egress = 3;
+
+            // no flows persisted -> we shouldn't have any interfaces
+            expectAllInterfaces();
+            expectIngressInterfaces();
+            expectEgressInterfaces();
+
+            // persist egress flow
```
Comment (fooker) [inline features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:283]: Comment / code mismatch (ingress vs egress)

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java [hunk 3]
```diff
@@ -221,11 +232,97 @@ public void testNMS12740() throws Exception {
...
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.INGRESS)), getMockFlowSource());
+            expectAllInterfaces(ingress);
+            expectIngressInterfaces(ingress);
+            expectEgressInterfaces();
+
+            // persist ingress flow
```
Comment (fooker) [inline features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:289]: And the other way around

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java [hunk 4]
```diff
@@ -108,6 +116,9 @@
     @Autowired
     private InterfaceToNodeCache interfaceToNodeCache;
 
+    @Autowired
+    private TransactionTemplate transactionTemplate;
```
Comment (fooker) [inline features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:120]: Use `SessionUtils` instead. It saves you from a lot of boilerplate code

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java [hunk 5]
```diff
@@ -221,11 +232,97 @@ public void testNMS12740() throws Exception {
...
+        factory.setHttpClientConfig(new HttpClientConfig.Builder("http://localhost:" + wireMockRule.port()).build());
+
+        try (JestClient client = factory.getObject()) {
+            final ElasticFlowRepository elasticFlowRepository = new ElasticFlowRepository(new MetricRegistry(),
+                    client, IndexStrategy.MONTHLY, documentEnricher,
+                    sessionUtils, nodeDao, snmpInterfaceDao,
+                    new MockIdentity(), new MockTracerRegistry(), new MockDocumentForwarder(), new IndexSettings(),
+                    mock(SmartQueryService.class));
+
+            Integer ingress = 2;
```
Comment (fooker) [inline features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:275]: Any reason why these (and others below) are `Integer`, not `int`?

Comment (fooker) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:275]: In this case: just inline them or make them real constants (same for the `nodeId` below.

**Review by fooker — CHANGES_REQUESTED** [review]

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java (hunk 4 above)
Comment (patrick-schweizer) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:120]: :+1:  fixed

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java (hunk 2 above)
Comment (patrick-schweizer) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:283]: fixed

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java (hunk 3 above)
Comment (patrick-schweizer) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:289]: fixed

File: features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java (hunk 5 above)
Comment (patrick-schweizer) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:275]: > Any reason why these (and others below) are `Integer`, not `int`?
nope. fixed.

Comment (patrick-schweizer) [reply features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java:275]: > In this case: just inline them or make them real constants (same for the `nodeId` below.

The reason I wanted to use a named variable (and not inline) is that it reads easier (2 has no meaning). 
But they are only used in the scope of the method so I didn't wanted to make it a class constant.
I made it final as a compromise. Sounds good?

**Review by christianpape — APPROVED** [review]
Looks good to me.

**Pull request merged**